src/ConnectionAutomaton.cpp

```diff
@@ -109,10 +121,14 @@ void ConnectionAutomaton::closeDuplexConnection(folly::exception_wrapper ex) {
     connectionOutput_.onComplete();
   }
   connectionInputSub_.cancel();
-  connection_.reset();
+}
 
-  stats_.socketClosed();
-  onClosed_();
+void ConnectionAutomaton::closeWithError(Frame_ERROR&& error) {
+  VLOG(4) << "closeWithError "
+          << error.payload_.data->cloneAsValue().moveToFbString();
+
+  outputFrameOrEnqueue(error.serializeOut());
+  close(folly::exception_wrapper(), StreamCompletionSignal::ERROR);
```

Comment:
Maybe everyone else apart from me was already clear that this is effectively NO_ERROR, instead of UNDEFINED_ERROR